### PR TITLE
Trac #15146 improve the UsersGuide for Lexically scoped type variables

### PR DESCRIPTION
**Pull request by AntC2**
Changes only to section 10.16.

Also reviewed references/links re ScopedTypeVariables from other sections: no changes needed.

### CODE REVIEW

File: docs/users_guide/glasgow_exts.rst [hunk 1]
```diff
@@ -9974,7 +10000,7 @@ example: ::
     f xs = (n, zs)
       where
         (ys::[a], n) = (reverse xs, length xs) -- OK
-        zs::[a] = xs ++ ys                     -- OK
+        (zs::[a]) = xs ++ ys                     -- OK
```
[the remark this answers is not in the document]
Comment (AntC2) [reply docs/users_guide/glasgow_exts.rst]: Done.

File: docs/users_guide/glasgow_exts.rst [hunk 2]
```diff
@@ -9809,6 +9809,17 @@ Lexically scoped type variables
     Enable lexical scoping of type variables explicitly introduced with
     ``forall``.
 
+.. tip::
+
+    ``ScopedTypeVariables`` breaks ghc's usual rule that explicit ``forall`` is optional and doesn't affect semantics.
```
[the remark this answers is not in the document]
Comment (AntC2) [reply docs/users_guide/glasgow_exts.rst]: Done.

File: docs/users_guide/glasgow_exts.rst [hunk 3]
```diff
@@ -9827,6 +9838,19 @@ signature for ``ys``. In Haskell 98 it is not possible to declare a type
 for ``ys``; a major benefit of scoped type variables is that it becomes
 possible to do so.
 
+An equivalent form for that example, avoiding explicit ``forall`` uses Pattern type signatures::
```
[the remark this answers is not in the document]
Comment (AntC2) [reply docs/users_guide/glasgow_exts.rst]: I was following the sub-title format. Have changed to a \:ref\:.

**Comment by AntC2** [comment]
Hmm. Seems I'm trying to do the impossible with that sub-heading: ReST doesn't like it, and that's by design [says StackOverflow](https://stackoverflow.com/questions/33149169/skipping-heading-levels-in-restructuredtext).
I tried just making it a bolded line of text, but got a big red stripe in the margin. Anybody any ideas?

**Comment by AntC2** [comment]
Changed to an in-line internal link, no heading: choose whichever commit is most aesthetic/in line with standards.

**Comment by bgamari** [comment]
Merged with 2ea93a7db7dcd9381b6d9e017ff015addd2adf5c.

**Pull request closed without merging**